server/src/main/java/org/elasticsearch/indices/IndicesServiceBuilder.java

```diff
@@ -177,6 +179,15 @@ public IndicesServiceBuilder mapperMetrics(MapperMetrics mapperMetrics) {
         return this;
     }
 
+    public List<SearchOperationListener> searchOperationListeners() {
+        return searchOperationListener;
+    }
+
+    public IndicesServiceBuilder searchOperationListeners(List<SearchOperationListener> searchOperationListener) {
+        this.searchOperationListener = searchOperationListener;
+        return this;
+    }
```

Comment:
Is it really needed to add this to the builder? Can't we just add the new listener unconditionally in `IndexModule`?

Reply:
We can do that - though that would mean to pass in a `MetricsRegistry` instance to the `IndexModule` so the new `ShardSearchAPMMetrics` could be built there. That felt more low-level than adding the listeners from the outside.

I thought it would be more general to add the listeners in `NodeConstruction` instead, which builds most of the telemetry related objects.

Happy to change that and pass along the `MetricsRegistry` if you think it's a better solution

Reply:
We need a single instance of the ShardSearchAPMMetrics for the entire node so your approach makes more sense. Let's keep it this way for now.